backend/src/question/question.service.ts

```diff
@@ -0,0 +1,186 @@
+   * @throws  NotFoundError
+   */
+  async getSummariesFromSlug(titleSlugs: string[]) {
+    const res: QuestionSummaryFromDb[] =
+      await this.prisma.questionSummary.findMany({
+        where: { titleSlug: { in: titleSlugs } },
+        select: { ...QUESTION_SUMMARY_SELECT },
+      });
+
+    return this.toQuestionSummaryTableType(res);
+  }
+
+  /**
+   * @return  {QuestionSummary}  Summary of Daily Question
+   */
+  async getDailyQuestionSummary() {
+    // Cron job ensures that there's only 1 QOTD at a time
+    const dailySummary: QuestionSummaryFromDb[] =
+      await this.prisma.questionSummary.findMany({
+        where: { isDailyQuestion: true },
+        select: { ...QUESTION_SUMMARY_SELECT },
+      });
+
+    return this.toQuestionSummaryTableType(dailySummary);
+  }
+
+  /**
+   * Gets a list of summaries that matches the given topics iff all of them are valid
+   * and intersect. Else, return an empty array.
+   *
+   * @param   {string[]}  topicTags  Array of topics to match
+   *
+   * @return  Array of matching, flattened QuestionSummary
+   */
+  async getSummariesFromTopicTags(
+    topicTags: string[]
+  ): Promise<FlattenedQuestionSummary[]> {
+    const res = await this.prisma.topicTag.findMany({
+      where: { topicSlug: { in: topicTags } },
+      select: { questionSummaries: { select: { titleSlug: true } } },
+    });
+
+    // consist of at least one invalid tag
+    if (topicTags.length != res.length) {
+      return [];
+    }
+
+    const slugArrays: string[][] = [];
+    for (const { questionSummaries } of res) {
+      slugArrays.push(questionSummaries.map((slug) => slug.titleSlug));
+    }
+
+    if (slugArrays.length === 1) {
+      return await this.getSummariesFromSlug(slugArrays.flat());
+    }
+
+    // Gets intersection of arrays (AND filter)
+    const intersect = slugArrays.reduce((prev, curr) =>
+      curr.filter(Set.prototype.has, new Set(prev))
+    );
+
+    return await this.getSummariesFromSlug(intersect);
+  }
+
+  async getAllTopics() {
+    const res = await this.prisma.topicTag.findMany({
+      select: { topicSlug: true },
+    });
+
+    return res.map((slug) => slug.topicSlug);
+  }
+
+  // ***** HELPER FUNCTIONS ***** //
+
+  /**
+   * Helper method to shape QuestionSummaryTableType into a consumer-friendly formay.
+   *
+   * @param   {QuestionSummaryFromDb[]}  data Raw data formatted by Prisma
+   *
+   * @return  {FlattenedQuestionSummary[]}  "Flattened" QuestionSummaryTableType
+   */
+  private toQuestionSummaryTableType(
+    data: QuestionSummaryFromDb[]
+  ): FlattenedQuestionSummary[] {
+    const normData = [
+      ...data.map((summary) => {
+        const { topicTags, updatedAt, titleSlug, ...info } = summary;
+        return {
+          ...info,
+          discussionLink: this.getDiscussionLink(titleSlug),
+          topicTags: topicTags.map((tag) => tag.topicSlug),
+          titleSlug,
+          updatedAt,
+        };
+      }),
+    ];
```

Comment:
think a more readable version would be - does the list destructuring do anyth special?
```suggestion
    const normData = data.map((summary) => {
      const { topicTags, updatedAt, titleSlug, ...info } = summary;
      return {
        ...info,
        discussionLink: this.getDiscussionLink(titleSlug),
        topicTags: topicTags.map((tag) => tag.topicSlug),
        titleSlug,
        updatedAt,
      };
    });
```